shared/util.js

```diff
@@ -325,6 +327,130 @@ class Util {
 
     return lines.filter(line => lineNumbersToKeep.has(line.lineNumber));
   }
+
+  /**
+   * Upgrades an lhr object in-place to account for changes in the data structure over major versions.
+   * @param {LH.Result} lhr
+   */
+  static upgradeLhrForCompatibility(lhr) {
```

Comment:
Maybe this should live in its own file? Seems like an easily describable utility function, it's already kind of long and pretty complex, and it's only going to grow. Better now and make just the one blame break.

Reply:
Done, no longer exposing from package entry. I believe this resolves your other comments here too.